main.py

```diff
@@ -0,0 +1,67 @@
+from fastapi import Depends, Query, FastAPI, HTTPException
+from sqlalchemy.orm import Session
+
+import crud
+import schemas
+from db.engine import SessionLocal
+
+app = FastAPI()
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.get("/authors/", response_model=list[schemas.Author])
+def read_authors_paginated(
+    page: int = Query(ge=0, default=0),
+    size: int = Query(ge=1, le=100),
+    db: Session = Depends(get_db),
+):
+    return crud.get_all_authors_paginated(db, page, size)
+
+
+@app.get("/authors/{author_id}/", response_model=schemas.Author)
+def read_single_author_by_id(author_id: int, db: Session = Depends(get_db)):
+    return crud.get_author_by_id(db, author_id)
+
+
+@app.post("/authors/", response_model=schemas.Author)
+def create_author(author: schemas.AuthorCreate, db: Session = Depends(get_db)):
+    if crud.get_author_by_name(db, author.name):
+        raise HTTPException(
+            status_code=400, detail="An author with such a name already exists"
+        )
+    return crud.create_author(db=db, author=author)
+
+
+@app.get("/books/", response_model=list[schemas.Book])
+def read_books_paginated(
+    author_id: int = None,
```

Comment:
Consider using `Optional[int]` from the `typing` module for the `author_id` parameter to explicitly indicate that it can be `None`.